File: Backend/src/main/java/com/lanit/lkz_project/entities/data_transfer_objects/PersonalAccountPageDto.java
package com.lanit.lkz_project.entities.data_transfer_objects;


import lombok.Data;


@Data
public class PersonalAccountPageDto<Notification> {

    private TimeFilter timeFilter;

    private boolean newFilter;

    private boolean inProcessingFilter;

    private boolean approvedFilter;

    private boolean rejectedFilter;

    private JsonPageImpl<Notification> page;

    private SortParameter sortParameter;

    private boolean reversedOrder;

}

File: Frontend/src/app/models/PersonalAccountPageDto.ts
import {JsonPageImpl} from './JsonPageImpl';
import {SortParameter} from './enums/SortParameter';
import {TimeFilter} from './enums/TimeFilter';


export interface PersonalAccountPageDto {

  timeFilter: TimeFilter;
  newFilter: boolean;
  inProcessingFilter: boolean;
  approvedFilter: boolean;
  rejectedFilter: boolean;
  page: JsonPageImpl;
  sortParameter: SortParameter;
  reversedOrder: boolean;

}
